src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -943,11 +943,32 @@ internal Type FastBindToType(string assemblyName, string typeName)
             if (entry == null || entry.AssemblyName != assemblyName)
             {
                 Assembly assm = null;
-                try
+                if (_isSimpleAssembly)
```

Comment:
Desktop calls LoadWithPartialNameInternal
https://referencesource.microsoft.com/#mscorlib/system/runtime/serialization/formatters/binary/binaryobjectreader.cs,1427 which does not exist in Core. After that point Core diverges. 

LoadWithPartialNameInternal calls nLoad(string...).  This change does Assembly.Load(string) instead. In Desktop, [Assembly.Load(string)](https://referencesource.microsoft.com/#mscorlib/system/reflection/assembly.cs,278) eventually calls to [nLoad](https://referencesource.microsoft.com/#mscorlib/system/reflection/assembly.cs,1828) with what seems to essentially be just the assembly name: so hopefully that is an equivalent substitution.